x/crisis/abci.go

```diff
@@ -12,7 +11,8 @@ import (
 
 // check all registered invariants
 func EndBlocker(ctx context.Context, k keeper.Keeper) {
-	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
+	start := telemetry.Now()
```

Comment:
this would take the time here and pass it into the defer later on, is that expected?

Reply:
thanks for the comment! if `time.Now()` is called within the `defer` statement, it takes the timestamp at the point where the `defer` is declared, not when the function exits. This leads to an inaccurate duration measurement because it doesn't reflect the actual time span of the operations we're interested in measuring, I realized that from @alexanderbez comment ([here](https://github.com/cosmos/cosmos-sdk/pull/19857#discussion_r1539571221)). By assigning `time.Now()` to start outside of defer, we ensure that we're measuring the duration of the function's operations accurately

Reply:
The statement is not correct. The parameters are evaluated when the function is deferred already and not on execution. See [demo](https://go.dev/play/p/A12MDMIux9W) or [stackoverflow](https://stackoverflow.com/questions/42703707/when-defer-func-evaluates-its-parameters)
Inlined or not is personal preference.

Reply:
💯 the parameters are evaluated and closed over when the function is deferred, not executed, I prefer the prior syntax (not using a local var) since it's fewer LoC.

Reply:
Thanks you both! So I'll use the prior syntax!